lib/types.nix

```diff
@@ -352,6 +352,25 @@ rec {
       functor = (defaultFunctor name) // { wrapped = [ t1 t2 ]; };
     };
 
+    coercedTo = coercedType: coerceFunc: finalType: mkOptionType rec {
+      name = "coercedTo";
+      description = "${finalType.description} or ${coercedType.description} coerced to it";
```

Comment:
These can be quite complex expressions, I am not sure that `coerce to it` would make sense in all cases.
The fact that some data are coerce is an implementation detail which does not matter from the documentation perspective of the description.  I would recommend using the same description as `types.either`.